lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java

```diff
@@ -1552,6 +1556,7 @@ private Task completeTask(String taskId, boolean isForced)
 
       Instant now = Instant.now();
       completeActionsOnTask(task, userId, now);
+      task = (TaskImpl) taskEndstatePreprocessorManager.processTaskBeforeEndstate(task);
```

Comment:
Please check if the SPI is enabled and only if that's the case execute this line of code